src/Simulation/Simulators/SparseSimulator/SparseSimulator.cs

```diff
@@ -34,16 +38,51 @@ public partial class SparseSimulator : CommonNativeSimulator
                randomNumberGeneratorSeed,
                disableBorrowing)
         {
-            Id = init_cpp((QubitIdType)numQubits);
+            // Init the telemetry first, to catch/log the construction issues:
+            TelemetryConfig = new TelemetryManagerConfig()
+            {
+                AppId = this.Name,
+                SendTelemetryInitializedEvent = true,
+                SendTelemetryTearDownEvent = true,
+            };
+            TelemetryManagerDisposer = TelemetryManager.Initialize(TelemetryConfig);
+
+            Microsoft.Applications.Events.EventProperties eventProperties = new Applications.Events.EventProperties()
+                {
+                    Name = "Quantum.SparseSim.Constructed",
+                };
+            eventProperties.SetProperty("throwOnReleasingQubitsNotInZeroState", throwOnReleasingQubitsNotInZeroState);
+            // (`SimulatorBase.Seed` is assigned to `randomNumberGeneratorSeed ?? Guid.NewGuid().GetHashCode()`):
+            eventProperties.SetProperty("randomNumberGeneratorSeed", Seed); 
```

Comment:
This seems a bit odd to me to collect; the only useful information I believe would be to know whether the seed has been set or not, but not which value.